src/pages/waiting-room/cat-c/waiting-room.cat-c.page.ts

```diff
@@ -0,0 +1,249 @@
+
+  pageState: WaitingRoomPageState;
+
+  formGroup: FormGroup;
+
+  subscription: Subscription;
+
+  merged$: Observable<boolean | string | CatBEUniqueTypes.JournalData>;
+
+  constructor(
+    public store$: Store<StoreModel>,
+    public navController: NavController,
+    public navParams: NavParams,
+    public platform: Platform,
+    public authenticationProvider: AuthenticationProvider,
+    private deviceAuthenticationProvider: DeviceAuthenticationProvider,
+    private screenOrientation: ScreenOrientation,
+    private insomnia: Insomnia,
+    private translate: TranslateService,
+    private modalController: ModalController,
+    private app: App,
+  ) {
+    super(platform, navController, authenticationProvider);
+    this.formGroup = new FormGroup({});
+  }
+
+  ionViewDidEnter(): void {
+    this.store$.dispatch(new waitingRoomActions.WaitingRoomViewDidEnter());
+
+    if (super.isIos()) {
+      this.screenOrientation.lock(this.screenOrientation.ORIENTATIONS.PORTRAIT_PRIMARY);
+      this.insomnia.keepAwake();
+    }
+
+    this.navBar.backButtonClick = (e: UIEvent) => {
+      this.clickBack();
+    };
+  }
+
+  clickBack(): void {
+    this.deviceAuthenticationProvider.triggerLockScreen()
+      .then(() => {
+        this.navController.pop();
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+
+  ngOnInit(): void {
+    const currentTest$ = this.store$.pipe(
+      select(getTests),
+      select(getCurrentTest),
+    );
+
+    this.pageState = {
+      insuranceDeclarationAccepted$: currentTest$.pipe(
+        select(getPreTestDeclarations),
+        select(getInsuranceDeclarationStatus),
+      ),
+      residencyDeclarationAccepted$: currentTest$.pipe(
+        select(getPreTestDeclarations),
+        select(getResidencyDeclarationStatus),
+      ),
+      signature$: currentTest$.pipe(
+        select(getPreTestDeclarations),
+        select(getSignatureStatus),
+      ),
+      candidateName$: currentTest$.pipe(
+        select(getJournalData),
+        select(getCandidate),
+        select(getCandidateName),
+      ),
+      candidateUntitledName$: currentTest$.pipe(
+        select(getJournalData),
+        select(getCandidate),
+        select(getUntitledCandidateName),
+      ),
+      candidateDriverNumber$: currentTest$.pipe(
+        select(getJournalData),
+        select(getCandidate),
+        select(getCandidateDriverNumber),
+        map(formatDriverNumber),
+      ),
+      welshTest$: currentTest$.pipe(
+        select(getJournalData),
+        select(getTestSlotAttributes),
+        select(isWelshTest),
+      ),
+      conductedLanguage$: currentTest$.pipe(
+        select(getCommunicationPreference),
+        select(getConductedLanguage),
+      ),
+    };
+    const { welshTest$, conductedLanguage$ } = this.pageState;
+    this.merged$ = merge(
+      currentTest$.pipe(
+        select(getJournalData),
+        tap((journalData: CatBEUniqueTypes.JournalData) => {
```

Comment:
If the above comment requires changes for CatCUniqueTypes (if any) this will need changed as well.